docs/proposals/ose-rebase.md

```diff
@@ -0,0 +1,120 @@
+moving all the changes in OSE that diverge from Origin into a set of commits that will sit
+on top of the Origin code[0] and a Jenkins job that will be rebasing master on a daily basis.
+Rebases are going to continue to happen only for the master branch.
+
+There are two types of commits that need to be carried on top of Origin:
+1. long-term carries related to branding and release tooling (master and release branches)
+2. CVEs and other backported bugfixes (release branches only)
+
+The Origin repository is set as a remote `upstream` branch in OSE so in master we can
+simply rebase upstream/master. Before the actual rebase happens, the last commit produced
+by tito during the previous rebase should be squashed into the main tito commit (one
+of the [CARRY] commits). If there is an intermediate commit that bumps the web console
+it should be dropped and a new updated may be added further down the build process. The
+tag that points to the last tito commit needs to be retaged because the squash is dropping
+it. If any conflicts are hit, manual resolution will be required as it is already the case
+with `git merge`. In the future, we can extend the conflict resolution mechanism in Jenkins
+to send an e-mail to the author and committer of the carry commits.
+
+A nightly build of OSE should follow the steps below in order to rebase on top of Origin:
+```sh
+git clone git@github.com/openshift/ose.git
+pushd ose
+git remote add upstream git@github.com/openshift/origin.git
+git fetch upstream
+git checkout master
+
+# Note down the current latest tag because squashing the tito commit into specfile updates
+# will drop it and tito tag needs it further down the build process in order to determine
+# the next (new) tag.
+last_tag="$( git describe --abbrev=0 --tags )"
+
+# sanity check that ensures the latest commits are in the right order.
+#
+# Either the last two commits are the tito tag commit and the specfile updates (where every
+# tito commit is eventually squashed):
+#
+#         .-Other [CARRY] commit
+#        /  .-Other [CARRY] commit
+#       /  /  .-[CARRY] Specfile updates
+#      /  /  /  .-Tito tag commit
+#     /  /  /  /
+# m--c--c--s--t
+#  \           \
+#   \           `-ose/master/HEAD,v3.5.0.23
+#    `-origin/master/HEAD
+#
+# or the last three commits are the tito tag commit, a web console bump that got created
+# during the previous rebase, and the specfile updates:
+#
+#         .-Other [CARRY] commit
+#        /  .-[CARRY] Specfile updates
+#       /  /  .-[DROP] webconsole bump
+#      /  /  /  .-Tito tag commit
+#     /  /  /  /
+# m--c--c--s--t
+#  \           \
+#   \           `-ose/master/HEAD,v3.5.0.23
+#    `-origin/master/HEAD
+#
+# In the first case, we simply want to reset the last two commits and commit them as one.
+# In the second case, we want the same thing as in the first case plus to completely drop
+# the webconsole bump since a new one may be created during the current build.
```

Comment:
What Steve said is correct.
origin-web-console/master => origin
origin-web-console/enterprise-3.5 => ose-3.5
origin-web-console/enterprise-3.4 => ose-3.4
...etc.
If the ose branding gets blown away, such as accidentally running the script pointing to master, and then you rerun the branding script, pointing to the correct branch, then the branding get's put back.

So, if the branding get's blown away, we just need to make sure it get's put back before the final tag.